File: src/tauTests/FileAndFolderManipulationTests.py
# -*- coding: utf-8 -*-
from Utilities import TestDataSource, Utils

class FileAndFolderManipulationTests(UITestCase):
    
    def setUp(self):
        global folderName1
        global folderName2
        global testingFolderName
        global amazeTestingFolder
        global fileName
        
        testingFolderName = TestDataSource.generalTestFolderName
        folderName1 = TestDataSource.folderName1
        folderName2 = TestDataSource.folderName2
        amazeTestingFolder = TestDataSource.amazeTestFolderName
        fileName = TestDataSource.textFileName
        launch.activity('com.amaze.filemanager', '.activities.MainActivity')
        
        Utils.navigateToTestFolder()
        Utils.createFolderWithName(amazeTestingFolder)
        tap.text(amazeTestingFolder)
        
    def tearDown(self):
        #packages.clearData('com.amaze.filemanager')
        #remove the folder
        Utils.navigateToTestFolder()
        tap.long.text(amazeTestingFolder)
        tap.resourceId('com.amaze.filemanager:id/delete')
        tap.resourceId('com.amaze.filemanager:id/buttonDefaultPositive')
        
        
        
        
        
    @testCaseInfo('<Add a new folder>', deviceCount=1)
    def testCreateNewFolderAndDeleteIt(self):
      
        #create the file
        #Utils.createFolderWithName(folderName1)
        #for some reason the way it is done in utils does not work!
        self.createFolderWithName(folderName1)
        
        #assert it is visible
        exists.text(folderName1)
        
        #delete the file
        tap.long.text(folderName1)
        tap.resourceId('com.amaze.filemanager:id/delete')
        tap.resourceId('com.amaze.filemanager:id/buttonDefaultPositive')
        
        #assert it is gone
        exists.no.text(folderName1)
        
    @testCaseInfo('<Add a new file>', deviceCount=1)
    def testCreateNewFileAndDeleteIt(self):
        
        #Utils.createFileWithName(fileName)
        self.createFileWithName(fileName)
        
        #assert file is visible
        exists.text(fileName)
        
        #delete the file
        tap.long.text(fileName)
        tap.resourceId('com.amaze.filemanager:id/delete')
        tap.resourceId('com.amaze.filemanager:id/buttonDefaultPositive')
        
        #assert it is gone
        exists.no.text(fileName)
        
        
    @testCaseInfo('<Copy a file to another folder>', deviceCount=1)
    def testCopyFileToAnotherFolder(self):
        self.createFolderWithName(folderName1)
        self.createFolderWithName(folderName2)
        
        #add file to the folder
        tap.text(folderName1)
        self.createFileWithName(fileName)
        
        #copy the file
        tap.resourceId("com.amaze.filemanager:id/properties")
        tap.text("Copy")
        
        input.key.back()
        Utils.swipeDownInPathBar()
        
        #paste the file to another folder
        tap.text(folderName2)
        tap.resourceId("com.amaze.filemanager:id/paste")
        
        #assert file is visible
        exists.text(fileName, wait=10000)
        
    
    def createFolderWithName(self, folderName):
        tap.instanceOf('android.widget.ImageView', index=1)
        tap.resourceId("com.amaze.filemanager:id/menu_item")
        input.text(folderName)
        tap.resourceId("com.amaze.filemanager:id/buttonDefaultPositive")
        
    def createFileWithName(self, name):
        tap.instanceOf('android.widget.ImageView', index=1)
        tap.resourceId("com.amaze.filemanager:id/menu_item1")
        input.text(name)
        tap.resourceId("com.amaze.filemanager:id/buttonDefaultPositive")
    

File: src/test/java/com/amaze/filemanager/test/TestClasses/FileAndFolderManipulationTests.java
package com.amaze.filemanager.test.TestClasses;

import com.amaze.filemanager.test.Utilities.TestDataSource;
import com.amaze.filemanager.test.Utilities.Utils;

import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import io.appium.java_client.TouchAction;

import static junit.framework.Assert.fail;

public class FileAndFolderManipulationTests extends BaseTestClass {

    private String fileName = TestDataSource.textFileName;
    private String folderName1 = TestDataSource.folderNames[0];
    private String folderName2= TestDataSource.folderNames[1];

    @Test
    public void testCreateNewFileAndDeleteIt(){
        try {
            //create the file
            Utils.createFileWithName(driver, fileName);

            //assert it exists
            if (Utils.findElementsByName(driver, fileName).isEmpty()) {
                fail("File is not visible");
            }

            //delete the file
            WebElement folder = stareAtPixies.until(ExpectedConditions.visibilityOfElementLocated(
                    By.xpath("//*[@text='" + fileName + "']")));
            TouchAction longPress = new TouchAction(driver);
            longPress.longPress(folder, 2000).release().perform();
            driver.findElementById("com.amaze.filemanager:id/delete").click();
            driver.findElementById("com.amaze.filemanager:id/buttonDefaultPositive").click();

            //assert the file is deleted
            if (!Utils.findElementsByName(driver, fileName).isEmpty()) {
                fail("File is still visible, should be deleted");
            }
        }
        catch(Exception e){
            takeScreenshot("failure_" + System.currentTimeMillis());
            throw e;
        }
    }

    @Test
    public void testCreateNewFolderAndDeleteIt(){
        try {
            //create the folder
            Utils.createFolderWithName(driver, folderName1);

            //assert it exists
            if (Utils.findElementsByName(driver, folderName1).isEmpty()) {
                fail("Folder is not visible");
            }

            //remove the folder
            WebElement folder = Utils.findElementByName(driver, folderName1);
            TouchAction longPress = new TouchAction(driver);
            longPress.longPress(folder, 2000).release().perform();
            driver.findElementById("com.amaze.filemanager:id/delete").click();
            driver.findElementById("com.amaze.filemanager:id/buttonDefaultPositive").click();

            //assert the folder is deleted
            if (!Utils.findElementsByName(driver, folderName1).isEmpty()) {
                fail("Folder is still visible, should be deleted");
            }
        }
        catch(Exception e){
            takeScreenshot("failure_" + System.currentTimeMillis());
            throw e;
        }
    }

    @Test
    public void testCopyFileToAnotherFolder() {
        try {
            Utils.createFolderWithName(driver, folderName1);
            Utils.createFolderWithName(driver, folderName2);

            //add file to folder 1
            Utils.findElementByName(driver, folderName1).click();
            Utils.createFileWithName(driver, fileName);

            //copy the file to the other folder
            driver.findElementById("com.amaze.filemanager:id/properties").click();
            stareAtPixies.until(ExpectedConditions.visibilityOfElementLocated(
                    By.xpath("//*[@text='Copy']")))
                    .click();

            driver.navigate().back();

            //have to swipe down to make the top part of toolbar visible
            Utils.swipeDownInPathBar(driver);

            //paste the file to another folder
            Utils.findElementByName(driver, folderName2).click();
            driver.findElementById("com.amaze.filemanager:id/paste").click();

            //wait for the slow copy action to perform
            WebDriverWait driverWait = new WebDriverWait(driver, 10);
            driverWait.until(ExpectedConditions.visibilityOfElementLocated(
                    By.xpath("//*[@text='" + fileName + "']")
            ));

            //assert the file is is visible
            if (Utils.findElementsByName(driver, fileName).isEmpty()) {
                fail("File should be pasted in, is not visible on the screen");
            }
        } catch (Exception e) {
            takeScreenshot("failure_" + System.currentTimeMillis());
            throw e;
        }

    }
}
